x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java

```diff
@@ -148,6 +153,13 @@ public String executorName() {
         return ThreadPool.Names.MANAGEMENT;
     }
 
+    @Override
+    public void beforeIndexCreated(Index index, Settings indexSettings) {
```

Comment:
I believe this should be `beforeIndexAddedToCluster` instead, since `beforeIndexCreated` is called on each data node in addition to the master (it isn't a huge deal, but might as well use the one better suited), `beforeIndexAddedToCluster` is only called on the master node.